feathr_project/feathr/registry/_feature_registry_purview.py

```diff
@@ -966,61 +1146,81 @@ def _list_registered_entities_with_details(self, project_name: str, entity_type:
             guid=guid_list)["entities"]
         return entity_res
 
-
     def get_features_from_registry(self, project_name: str) -> Tuple[List[FeatureAnchor], List[DerivedFeature]]:
         """Sync Features from registry to local workspace, given a project_name, will write project's features from registry to to user's local workspace]
         If the project is big, the return result could be huge.
         Args:
             project_name (str): project name.
         """
-        all_entities_in_project = self._list_registered_entities_with_details(project_name=project_name,entity_type=[TYPEDEF_DERIVED_FEATURE, TYPEDEF_ANCHOR_FEATURE, TYPEDEF_FEATHR_PROJECT, TYPEDEF_ANCHOR, TYPEDEF_SOURCE])
-        if not all_entities_in_project:
+        project_entity = self.purview_client.get_entity(qualifiedName=project_name,typeName=TYPEDEF_FEATHR_PROJECT)['entities'][0]
+        
+        all_lineages = self.purview_client.get_entity_lineage(project_entity['guid'])
+        single_direction_process = [x for _,x in self.purview_client.get_entity_lineage(project_entity['guid'])['guidEntityMap'].items() \
```

Comment:
Should we change `self.purview_client.get_entity_lineage(project_entity['guid']` to `all_lineages ` to avoid additional calls?

Reply:
removed all_lineages